js/app.js

```diff
@@ -78,6 +78,7 @@ function getParameterByNameBlocko(name, url) {
 app.controller('CheckoutController', function($scope, $interval, Order, $httpParamSerializer, $timeout, AltcoinNew, AltcoinAccept, AltcoinLimits, WpAjax) {
     //get order id from url
     $scope.address = getParameterByNameBlocko("show_order");
+	$scope.refresh = getParameterByNameBlocko("refresh");
```

Comment:
I don't think a refersh parameter is needed. This causes a problem because if customer presses F5 on payment screen. It will again restart the payment timer, which is not a desired behaviour.

What we need to do is to do a call to backend to update order of timestamp/BTC value of order (according to latest price) and then just reload the *show_order* page. It should automatically work

Reply:
Modified with the newest commit to the branch.

Reply:
We should remove the reset parameter from the url as this will cause a new address to be generated on consecutive refreshes.  I recommend using the timestamp to check if an order is expired and generate a new timestamp and BTC value if required.

Reply:
reset parameter, as per Blockonomics API, won't generate new address (it's what we sent to newAddress function) as in the 'pay with bitcoin' page we didn't have access to the current address. And as process_payment is only used when we click 'pay with bitcoins' function (overridden from woocommerce i assume), that's where we should do the check. 

Should i make another option to save up current address? Timestamp call needs address too.

Reply:
OK, pushed new commit. This time using timestamp for checking, and order_id as index from blockonomics_orders.